core/interfaces/i_dragger.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Coordinate} from '../utils/coordinate';
+
+export interface IDragger {
+  /**
+   * Handles any drag startup.
+   *
+   * @param e PointerEvent that started the drag.
+   */
+  onDragStart(e: PointerEvent): void;
+
+  /**
+   * Handles calculating where the element should actually be moved
+   * to.
+   *
+   * @param e PointerEvent that continued the drag.
+   * @param totalDelta Cumulative drag distance, in workspace
```

Comment:
Currently the equivalent function uses pixel units, not workspace units. That is probably what we want so we can support scrolling on dragging to the edge of the workspace in the scroll options plugin.

The reason we pass the total delta is because we don't actually start a drag until the mouse has already moved a certain distance, (e.g. we need to make sure when dragging from the fly out we're actually dragging toward the workspace). But we want the `IDraggable` to catch up with the mouse (otherwise it would be offset from where the user originally grabbed the `IDraggable`).

Reply:
JSDocs updated accordingly.